### PR TITLE
Rewrite enumeration using DirectoryInfo and LINQ

### PR DESCRIPTION
**Pull request by ricardoboss**

### CODE REVIEW

File: TRENZ.Docs.API/Models/Sources/AbstractFilesystemSource.cs [hunk 1]
```diff
@@ -1,4 +1,5 @@
 ﻿using System.Text.RegularExpressions;
+using IOPath = System.IO.Path;
```
Comment (chucker) [inline TRENZ.Docs.API/Models/Sources/AbstractFilesystemSource.cs]: Do we have a conflict?

Maybe `FSPath` for `FileSystemPath`?

File: TRENZ.Docs.API/Models/Sources/AbstractFilesystemSource.cs [hunk 2]
```diff
@@ -28,29 +29,18 @@ public IEnumerable<SourceFile> FindFiles(Regex pattern)
...
-
-            if (!Directory.Exists(file))
-                continue;
-
-            foreach (var f in IterateDirectory(pattern, file, root))
-            {
-                yield return f;
-            }
-        }
+            AttributesToSkip = FileAttributes.System,
```
Comment (chucker) [inline TRENZ.Docs.API/Models/Sources/AbstractFilesystemSource.cs]: Add comment here:

    // don't skip hidden files such as `.order`

**Comment by chucker** [comment]
Should be fine since we flatten the hierarchy anyway

File: TRENZ.Docs.API/Models/Sources/AbstractFilesystemSource.cs (hunk 1 above)
Comment (ricardoboss) [reply TRENZ.Docs.API/Models/Sources/AbstractFilesystemSource.cs]: `AbstractFilesystemSource` has an auto-property named `Path`, yes.

I prefer `FileSystemPath`, actually. But it's quite long too...

File: TRENZ.Docs.API/Models/Sources/AbstractFilesystemSource.cs [hunk 3]
```diff
@@ -28,29 +29,18 @@ public IEnumerable<SourceFile> FindFiles(Regex pattern)
...
     {
-        foreach (var file in Directory.EnumerateFileSystemEntries(path, "**", new EnumerationOptions
-                 {
-                     AttributesToSkip = FileAttributes.System,
-                     IgnoreInaccessible = true,
-                     MatchCasing = MatchCasing.CaseInsensitive,
-                     MatchType = MatchType.Win32,
-                     RecurseSubdirectories = true,
-                 }).Where(f => pattern.IsMatch(System.IO.Path.GetFileName(f))))
+        var info = new DirectoryInfo(path);
```
Comment (chucker) [inline TRENZ.Docs.API/Models/Sources/AbstractFilesystemSource.cs]: Can we call this `dir`. `info.EnumerateFiles()` is weird.

File: TRENZ.Docs.API.Test/LocalSourceTest.cs [hunk 4]
```diff
@@ -70,4 +70,21 @@ public void TestToString(Dictionary<string, string> config, string expected)
         var source = new LocalSource(configuration);
         Assert.AreEqual(expected, source.ToString());
     }
+
+    public static IEnumerable<object[]> FindFilesValuesProvider()
+    {
+        yield return new object[] { new Dictionary<string, string> { { "Type", "local" }, { "Name", "Test" }, { "Root", "./Data/" } }, new[] { "Image", "Test" } };
```
Comment (chucker) [inline TRENZ.Docs.API.Test/LocalSourceTest.cs]: Hmm,

* the test `TestFindFiles` would be more useful with some additional edge cases. (For example, test if it finds a `.order` file.)
* I would also prefer if the items here are vertical so that it's easier to see that you're creating a hierarchy.

**Review by chucker — APPROVED** [review]
